graylog2-server/src/main/java/org/graylog2/periodical/ContentPackLoaderPeriodical.java

```diff
@@ -89,37 +112,112 @@ protected Logger getLogger() {
 
     @Override
     public void doRun() {
-        final Set<String> resources = new Reflections(RESOURCE_PREFIX, new ResourcesScanner()).getResources(RESOURCE_PATTERN);
+        final ContentPackLoaderConfig contentPackLoaderConfig =
+                clusterConfigService.getOrDefault(ContentPackLoaderConfig.class, ContentPackLoaderConfig.EMPTY);
+
+        final List<Path> files = getFiles(contentPacksDir, FILENAME_GLOB);
+        final Map<String, ConfigurationBundle> contentPacks = new HashMap<>(files.size());
+        for (Path file : files) {
+            final String fileName = file.getFileName().toString();
+            if (contentPackLoaderConfig.loadedContentPacks().contains(fileName)) {
+                LOG.debug("Skipping already loaded content pack {}", file);
+                continue;
+            }
 
-        for (String resource : resources) {
-            final URL resourceUrl = Resources.getResource(resource);
-            final byte[] resourceBytes;
+            LOG.debug("Reading content pack from {}", file);
+            final byte[] bytes;
             try {
-                resourceBytes = Resources.toByteArray(resourceUrl);
+                bytes = Files.readAllBytes(file);
             } catch (IOException e) {
-                LOG.warn("Couldn't read " + resource + ". Skipping.", e);
+                LOG.warn("Couldn't read " + file + ". Skipping.", e);
                 continue;
             }
 
+            LOG.debug("Parsing content pack from {}", file);
             final ConfigurationBundle contentPack;
             try {
-                contentPack = objectMapper.readValue(resourceBytes, ConfigurationBundle.class);
+                contentPack = objectMapper.readValue(bytes, ConfigurationBundle.class);
             } catch (IOException e) {
-                LOG.warn("Couldn't parse content pack " + resource + ". Skipping", e);
+                LOG.warn("Couldn't parse content pack in file " + file + ". Skipping", e);
+                continue;
+            }
+
+            final ConfigurationBundle existingContentPack = contentPackExists(contentPack);
```

Comment:
As a matter of fact the method returned a boolean before but was refactored to return the actual content pack. :wink: